### PR TITLE
refactor(ci): Bypass Lerna and repurpose the 'test-travis' run-script

### PR DESCRIPTION
**Pull request by spencerwilson-optimizely**
## Summary

- Bypass Lerna things in CI
- Don't install `grunt`. I presume it was installed because `test-travis` refers to it, but locally-installed executable modules are on the $PATH in run-scripts so global installation is unnecessary.
- Rename `test-travis` to `test-xbrowser` and modify it such that it only runs cross-browser tests in BrowserStack (via `grunt`), rather than re-running unit tests headlessly+locally _again_ with `npm test` (which is done as part of the `prepare` lifecycle hook that's invoked during `npm install`). The rename makes it clearer that the run-script could plausibly, usefully be run locally as well (say, if one was debugging a consistent failure in some browser).

Lerna seems to be correlated with some recent sporadic timeout failures:
- https://travis-ci.org/optimizely/javascript-sdk/jobs/397898591
- https://travis-ci.org/optimizely/javascript-sdk/jobs/397898592
- https://travis-ci.org/optimizely/javascript-sdk/jobs/397568472
- https://travis-ci.org/optimizely/javascript-sdk/jobs/397568474
- https://travis-ci.org/optimizely/javascript-sdk/jobs/396745267

This PR `cd`'s the Travis CI pipeline directly into the optimizely-sdk package's dir, and does things there. Doing so seems to reduce the mean CI time (compute time on Travis machines, not wall-clock time) from about 21 mins to 15 mins.

## Test plan
This PR's CI run: [#288](https://travis-ci.org/optimizely/javascript-sdk/builds/397977186).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,12 +15,9 @@ env:
...
   # BROWSER_STACK_ACCESS_KEY
   - secure: U0GGZw46rJowBtH9gVluIrerB40u2b3uZpH0HsOdLlsXCCaTVk4JXX/JPVPashWAFLC7Enk3UOE4ofeEpVd0wbG6CxtG9/gklc2U2tvkqsdPpFZKaRrXoUzCyyPOmHEC2mXDXctbrncmttM4APaceRfbdTBEZIIfyLJadomjWylA61szFE9IZjvJpiwJO2xa5HI9GVRu3yXJci+riJux+JsDmfJ1hNwv3waMeeg/scddUH0hfgq69ftGs8cpMlYiO20eh32S7uPF7/IJTH1fDJjVKYQZwpypkF6AeI+od5CFTY1ajb25eaBNXThLS0Bo9ZJE/8Sogvon21dEJkt/ClY6R341InbAFXZvz7jyQAisvh0I4zxcu0VUCfh7bEUl6GXMO8VJnyxHEfqB+AIT2RoMXckkhulwiNUsJYH1yJ8mjnLvZq85mWBCp4n4jg0K6Wf46lHpjnHOVpLyLyoFGfiPf90AQVL02AJ3/ia8RkMuj0Ax+AGtiTC/+wy7dsDQOif/VpBNJcx/RciQ24mYOGzAMh4GsUWnXaZ9vXSxliogVNrmIefK5invJ0omv9pIx8NZHTHYGaulh4w6JsliiEq2kH78SlyvSrcsFGTwCY97LLaxiLm/75/Zf+F7LajKC23Fbtnj/LQizitFZqGMJ09DnR52krBAeultqRq8QLM=
-before_script:
-- npm install grunt-cli -g
```
Comment (tylerbrandt) [inline .travis.yml]: not necessary anymore?

Comment (spencerwilson-optimizely) [reply .travis.yml]: Never necessary, I think. Added an explanation to the PR summary.

Comment (tylerbrandt) [reply .travis.yml]: 👍 I wonder how much instability this added. I definitely saw on at least one build that the "hang" occurred while installing `grunt-cli`.

File: packages/optimizely-sdk/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "browser": "lib/index.browser.js",
   "scripts": {
     "test": "mocha ./lib/*.tests.js ./lib/**/*.tests.js ./lib/**/**/*tests.js --recursive",
-    "test-travis": "npm run test && grunt",
+    "test-xbrowser": "grunt",
```
Comment (tylerbrandt) [inline packages/optimizely-sdk/package.json]: nit: It's a little weird that the "default" grunt command is one that runs xbrowser tests specifically; I might use the specific named grunt task here, if there is one.

Comment (spencerwilson-optimizely) [reply packages/optimizely-sdk/package.json]: I found it surprising too. It's defined as the default task in Gruntfile.js.

**Comment by spencerwilson-optimizely** [comment]
> I have a mild preference to keep using `lerna` if we aren't sure it is actually the cause of the problem, just to avoid having so much unused stuff in this repo

I hear you. I'm still kind of swayed by how bypassing lerna reduces build times. That's valuable to the extent that (at least as of 2015) [open source build concurrency is limited to 5](https://twitter.com/travisci/status/651856122559774722). 🤷‍♂️  Though that argument is weakened, I think, by the fact that when this repo moves to travis-ci.com, concurrency can be boosted because it'll be linked with our paid Travis CI account:
> open source projects will be joining private projects on travis-ci.com ... This means you can ... allocate additional concurrency to either open source or private projects ...

([source](https://docs.travis-ci.com/user/open-source-on-travis-ci-com/))

**Pull request merged**